From eafa3b5617bcefeb19e6112ee869d912667d26bb Mon Sep 17 00:00:00 2001
From: Chris Packham <chris.packham@alliedtelesis.co.nz>
Date: Thu, 30 Oct 2025 17:12:03 +1300
Subject: [PATCH] use stdbool

Use stdbool to provide a bool type. A similar change was done are in commit
ea780522e ("gnulib: Drop fix-base64.patch") but this won't apply cleanly to
2.04 so this change brings in only the stdbool related changes.
---
 grub-core/lib/posix_wrap/sys/types.h | 6 ++----
 grub-core/lib/xzembed/xz.h           | 5 +----
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/grub-core/lib/posix_wrap/sys/types.h b/grub-core/lib/posix_wrap/sys/types.h
index 854eb0122..a13f5edb5 100644
--- a/grub-core/lib/posix_wrap/sys/types.h
+++ b/grub-core/lib/posix_wrap/sys/types.h
@@ -23,11 +23,9 @@
 
 #include <stddef.h>
 
+#include <stdbool.h>
+
 typedef grub_ssize_t ssize_t;
-#ifndef GRUB_POSIX_BOOL_DEFINED
-typedef enum { false = 0, true = 1 } bool;
-#define GRUB_POSIX_BOOL_DEFINED 1
-#endif
 
 typedef grub_uint8_t uint8_t;
 typedef grub_uint16_t uint16_t;
diff --git a/grub-core/lib/xzembed/xz.h b/grub-core/lib/xzembed/xz.h
index f7b32d800..d1417039a 100644
--- a/grub-core/lib/xzembed/xz.h
+++ b/grub-core/lib/xzembed/xz.h
@@ -29,10 +29,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <grub/misc.h>
-
-#ifndef GRUB_POSIX_BOOL_DEFINED
-typedef enum { false = 0, true = 1 } bool;
-#endif
+#include <stdbool.h>
 
 /**
  * enum xz_ret - Return codes
